tests/client/test_async_client.py

```diff
@@ -155,3 +156,10 @@ async def test_100_continue(server):
 
     assert response.status_code == 200
     assert response.content == data
+
+
+@pytest.mark.usefixtures("async_environment")
+async def test_request_with_none_header(server):
+    value: Optional[str] = None
+    async with httpx.AsyncClient() as client:
+        await client.get(server.url, headers={"foo": value})
```

Comment:
(Nit) It looks like this test is redundant with the one already added in `test_headers.py`. I think we can drop it. (The `typing` important can then be removed too.)

```suggestion
```